src/listener/handlers/document-update-handler.ts

```diff
@@ -107,37 +136,49 @@ class DocumentUpdateHandler implements IHandler {
+                reject(err);
+            });
         });
     }
 
     /**
      * check if the document is a File and it is VALID, if so refresh the doc and check for an assigned cloudUrl.
      * proceed with file validation is cloudUrl present, other invalidate file validationState
-     * @param contentValidationReport 
-     * @param documentUrl 
-     * @param documentType 
+     * @param contentValidationReport
+     * @param documentUrl
+     * @param documentType
      */
-    checkEligibleForFileValidation(contentValidationReport: ValidationReport, documentUrl: string, documentType: string) : Promise<ValidationReport> {
-        if(documentType.toUpperCase() === 'FILE') {
-            if (contentValidationReport.validationState.toUpperCase() == "VALID") {
-                // refresh doc before checking cloudUrl
-                return this.ingestClient.retrieveMetadataDocument(documentUrl).then((doc: any) => {
-                    if (doc['cloudUrl'] ) {
-                        // if cloud url present, proceeds with file validation
-                        return this.attemptFileValidation(contentValidationReport, doc, documentType) 
-                    } else {
-                        // otherwise set validation state to INVALID, return error report with NoCloudUrl
-                        const msg = "File cloudUrl property not set.";
-                        const err = new ErrorReport(ErrorType.FileNotUploaded, msg);
-                        err.userFriendlyMessage = "File not uploaded.";
-                        return Promise.resolve(new ValidationReport("INVALID", [err]));
+    checkEligibleForFileValidation(contentValidationReport: ValidationReport, documentUrl: string, documentType: string): Promise<ValidationReport> {
+        if (documentType.toUpperCase() === 'FILE') {
+            // refresh doc before checking cloudUrl
+            return this.ingestClient.retrieveMetadataDocument(documentUrl).then((doc: any) => {
+                if (doc['cloudUrl']) {
+                    // if cloud url present, proceeds with file validation
+                    if (contentValidationReport.validationState.toUpperCase() == "VALID") {
+                        return this.attemptFileValidation(contentValidationReport, doc, documentType)
                     }
 
-                })
-            }
+                } else {
+                    // otherwise set validation state to INVALID, return error report with NoCloudUrl
+                    const msg = "File cloudUrl property not set.";
+                    const err = new ErrorReport(ErrorType.FileNotUploaded, msg);
+                    err.userFriendlyMessage = "File not uploaded.";
+
+                    contentValidationReport.validationState = "INVALID";
+
+                    if (!contentValidationReport.validationErrors || contentValidationReport.validationErrors.length == 0) {
+                        contentValidationReport.validationErrors = [];
+                    }
```

Comment:
is this really needed when it we were previously setting (overriding) an validation errors anyway using
`new ValidationReport("INVALID", [err])`

Reply:
ah this check is to just make sure contentValidationReport.validationErrors is not undefined when i do 
```
contentValidationReport.validationErrors.push(err)
```
actually i might not need the second condition

I need to append the "file not uploaded" error to the metadata errors.